src/videouplinkbandwidthpolicy/NScaleVideoUplinkBandwidthPolicy.ts

```diff
@@ -167,27 +167,21 @@ export default class NScaleVideoUplinkBandwidthPolicy implements VideoUplinkBand
     if (!settings) {
       return;
     }
-
     const encodingParams: RTCRtpEncodingParameters = this.calculateEncodingParameters(settings);
-    if (
-      !this.encodingParametersHaveSameBitrateAndScaling(
-        encodingParams,
-        this.encodingParamMap.get(NScaleVideoUplinkBandwidthPolicy.encodingMapKey)
-      )
-    ) {
+    if (this.shouldUpdateEndcodingParameters(encodingParams)) {
       this.encodingParamMap.set(NScaleVideoUplinkBandwidthPolicy.encodingMapKey, encodingParams);
       this.transceiverController.setEncodingParameters(this.encodingParamMap);
     }
   }
 
-  private encodingParametersHaveSameBitrateAndScaling(
-    encoding1: RTCRtpEncodingParameters,
-    encoding2: RTCRtpEncodingParameters
-  ): boolean {
+  private shouldUpdateEndcodingParameters(encoding: RTCRtpEncodingParameters): boolean {
+    /* istanbul ignore next: sender, getParameters, and encodings are optional */
+    const transceiverEncoding = this.transceiverController
```

Comment:
It is the same thing but can we loop through the encoding array and check the key since we do the same in transceiver controller.

Reply:
Could you elaborate more on how to check the key?

https://github.com/aws/amazon-chime-sdk-js/blob/4d779e41136d927da7816e38c5c8cd6535f40897/src/transceivercontroller/DefaultTransceiverController.ts#L37-L56

Reply:
I think you can just check if `encoding[i].rid === encodingMapKey`.

Reply:
The value of `encodingMapKey` is `video`, but `rid` is not available.

```
active: true
adaptivePtime: false
maxBitrate: 601000
networkPriority: "low"
priority: "low"
scaleResolutionDownBy: 1
```

1. NScale passes a map of <encodingMapKey, object>.
   https://github.com/aws/amazon-chime-sdk-js/blob/4d779e41136d927da7816e38c5c8cd6535f40897/src/videouplinkbandwidthpolicy/NScaleVideoUplinkBandwidthPolicy.ts#L179
2. The transceiver controller only uses the values but not `encodingMapKey`.
   https://github.com/aws/amazon-chime-sdk-js/blob/4d779e41136d927da7816e38c5c8cd6535f40897/src/transceivercontroller/DefaultTransceiverController.ts#L31

Reply:
My bad. We only use that as a set in the video policy itself, the encoding params does not have it.